--- postgresql-16.2/src/port/path.c	2024-02-05 22:41:37.000000000 +0100
+++ postgresql-16.2-wasm/src/port/path.c	2024-04-24 08:49:36.573534270 +0200
@@ -598,6 +598,15 @@
 	if (strlen(progname) > sizeof(EXE) - 1 &&
 		pg_strcasecmp(progname + strlen(progname) - (sizeof(EXE) - 1), EXE) == 0)
 		progname[strlen(progname) - (sizeof(EXE) - 1)] = '\0';
+#elif defined(__EMSCRIPTEN__)
+	/* remove .js and make log a commment */
+	if (progname && strlen(progname)>4) {
+		for (int i=strlen(progname)-2;i>2;i--)
+			progname[i] = progname[i-2];
+		progname[0]= '#';
+		progname[1]= ' ';
+		progname[strlen(progname)-1] = '\0';
+	}
 #endif
 
 	return progname;
