--- postgresql-16.2/src/backend/commands/collationcmds.c	2024-02-05 22:41:37.000000000 +0100
+++ postgresql-16.2-wasm/src/backend/commands/collationcmds.c	2024-04-24 09:12:16.035273102 +0200
@@ -828,8 +828,11 @@
 		maxaliases = 100;
 		aliases = (CollAliasData *) palloc(maxaliases * sizeof(CollAliasData));
 		naliases = 0;
-
+#if !defined(__EMSCRIPTEN__) && !defined(__wasi__)
 		locale_a_handle = OpenPipeStream("locale -a", "r");
+#else
+		locale_a_handle = fopen(WASM_PREFIX "/locale","r");
+#endif
 		if (locale_a_handle == NULL)
 			ereport(ERROR,
 					(errcode_for_file_access(),
